compiler/rustc_typeck/src/check/wfcheck.rs

```diff
@@ -808,7 +808,9 @@ fn check_param_wf(tcx: TyCtxt<'_>, param: &hir::GenericParam<'_>) {
 
         // Const parameters are well formed if their type is structural match.
         hir::GenericParamKind::Const { ty: hir_ty, default: _ } => {
-            let ty = tcx.type_of(tcx.hir().local_def_id(param.hir_id));
+            let def_id = tcx.hir().local_def_id(param.hir_id);
+            let ty = tcx.type_of(def_id);
+            let ty = tcx.normalize_erasing_regions(tcx.param_env(def_id), ty);
```

Comment:
this normalize is suspicious, though we don't have an inferctxt to use for normalization here... we might need to write a separate query just to "peel" type aliases. (might need to use it for the simd code above, too? probably not, though.)

Reply:
This normalize was not needed. I removed it.